src/components/lib/cos_kernel_api.c

```diff
@@ -200,7 +201,7 @@ __capid_captbl_check_expand(struct cos_compinfo *ci)
 
 	printd("__capid_captbl_check_expand->pre-captblactivate (%d)\n", CAPTBL_OP_CAPTBLACTIVATE);
 	/* captbl internal node allocated with the resource provider's captbls */
-	if (call_cap_op(meta->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, captblcap, meta->pgtbl_cap, kmem, 1)) {
+	if (call_cap_op(meta->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, captblcap, meta->local_mi.pgtbl_cap, kmem, 1)) {
```

Comment:
`captblcap` here is the new capability id that was created at the `frontier` not referring to our own captbl. That which was just activated at this step, will need to be `OP_CONS`ed into the `ci->captbl_cap` where we are trying to extend at `captblid_add`. I can see that there is no consistency check for this address in the kernel on `OP_CONS`, I wonder if that is required.